mne_connectivity/decoding/decomposition.py

```diff
@@ -324,19 +334,10 @@ def _check_X(self, X, ndim):
     def _get_rank_and_ncomps_from_X(self, X):
         """Get/validate rank and n_components parameters using the data."""
         # compute rank from data if necessary / check it is valid for the indices
-        rank = _check_rank_input(self._rank, X, self._indices)
+        self._rank = _check_rank_input(self._rank, X, self._indices)
 
-        # set n_components if necessary / check it is valid for the rank
-        if self.n_components is None:
-            self.n_components = np.min(rank)
-        elif self.n_components > np.min(rank):
-            raise ValueError(
-                "`n_components` is greater than the minimum rank of the data"
-            )
-
-        # set rank if necessary
-        if self._rank is None:
-            self._rank = rank
+        # check n_components is valid for the rank
+        self.n_components = _check_n_components_input(self.n_components, self._rank)
```

Comment:
Little refactoring change since this check is required also in `spec_conn_epochs` and `spec_conn_time` now